src/main/java/org/broadinstitute/hellbender/tools/spark/CreateIndexForHadoopBam.java

```diff
@@ -0,0 +1,158 @@
+package org.broadinstitute.hellbender.tools.spark;
+
+import htsjdk.samtools.BAMIndex;
+import htsjdk.samtools.BAMIndexer;
+import htsjdk.samtools.BamFileIoUtils;
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.SamReader;
+import htsjdk.samtools.SamReaderFactory;
+import htsjdk.samtools.ValidationStringency;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.cmdline.CommandLineProgram;
+import org.broadinstitute.hellbender.cmdline.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.SparkProgramGroup;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.io.IOUtils;
+import org.broadinstitute.hellbender.utils.read.ReadConstants;
+import org.codehaus.plexus.util.FileUtils;
+import org.seqdoop.hadoop_bam.SplittingBAMIndexer;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+@CommandLineProgramProperties(summary = "Create a HadoopBam splitting index as well as a bam index from a bam file.",
+        oneLineSummary = "create a hadoop-bam splitting index" ,
+        programGroup = SparkProgramGroup.class)
+public final class CreateIndexForHadoopBam extends CommandLineProgram {
```

Comment:
renamed.  I'm not sure how spark tools can detect if there is a splitting index being used or not.  I opened a ticket to emit this warning #1563